charts/prometheus-alerts/templates/containers-prometheusrule.yaml

```diff
@@ -200,6 +200,34 @@ spec:
         {{- end }}
     {{- end }}
 
+    {{ with .PodSelectorValidity -}}
+    - alert: PodSelectorValidity
+      annotations:
+        summary: PodSelector for prometheus-alerts is invalid
+        runbook_url: {{ $.Values.defaults.runbookUrl }}#Alert-Rules-Selectors-Validity
+        description: >-
+          The PodSelector used for pod level alerts did not return any data.
+          Please check the PodSelector applied in your prometheus-alerts chart
+          is correct to ensure you are properly selecting your pods so that you
+          will be alerted for pod issues. The current selector is
+          `{podSelector="{{ $podSelector }}", namespaceSelector="{{ $namespaceSelector }}"}`.
```

Comment:
I feel like this string is wrong ... shouldn't it just be `{ {{ $podSelector }}, {{ $namespaceSelector }} }`?

Reply:
I was not trying to put the exact promql. I was trying to map the prometheus-alerts values name to the rendered output to give people a searchable string in the values file and the actual promql.